cfg_mgmt/github.py

```diff
@@ -0,0 +1,326 @@
+import dataclasses
+import datetime
+import logging
+import os
+import typing
+import yaml
+
+import git
+from Crypto.PublicKey import RSA
+
+import ccc.github
+import cfg_mgmt.model as cmm
+import cfg_mgmt.util
+import ci.util
+import concourse.util
+import gitutil
+import model
+import model.github
+
+from cfg_mgmt.model import (
+    CfgQueueEntry,
+    CfgTarget,
+    CfgStatus,
+    cfg_queue_fname,
+    cfg_status_fname,
+)
+from model.github import (
+    GithubConfig,
+)
+
+
+ci.log.configure_default_logging()
+logger = logging.getLogger(__name__)
+
+
+def rotate_github_config(
+    cfg_dir: str,
```

Comment:
we added a new parameter `cfg_factory` to `rotate_cfg_element` yesterday; -> you might want to follow suit and also pass-in cfg-factory to this function